### PR TITLE
Imports: implement contiguousGroups logic

### LINKED ISSUE #2718: Sort imports with groups can reorder program execution
This template is a guideline, not a strict requirement.
- **Version**: 3.0.1
- **Integration**: IntelliJ
- **Configuration**:
```
version = "3.0.1"
rewrite.rules = [Imports]
rewrite.imports.groups = [["""scala\..*"""]]
```

## Steps

Given code like this:
```scala
object Main {
  import Foo._
  val bar: Bar = getBar
  import bar.ec

  def getBar: Foo.Bar = ???
}
object Foo {
  trait Bar {
    implicit def ec: ExecutionContext
  }
}
```

When I run scalafmt like this:
IntelliJ keybind

## Problem

Scalafmt formats code like this:
```scala
object Main {
  import Foo._
  import bar.ec
  val bar: Bar = getBar

  def getBar: Foo.Bar = ???
}
object Foo {
  trait Bar {
    implicit def ec: ExecutionContext
  }
}
```

## Expectation

I would like the formatted output to look like this:
```scala
object Main {
  import Foo._
  val bar: Bar = getBar
  import bar.ec

  def getBar: Foo.Bar = ???
}
object Foo {
  trait Bar {
    implicit def ec: ExecutionContext
  }
}
```

## Workaround
Keeping the first import outside the class it works fine.

## Notes

See also..

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #2718.

### CODE REVIEW

**Comment by kitbellew** [comment]
@tgodzik thoughts on this?

File: docs/configuration.md [hunk 1]
```diff
@@ -2668,6 +2659,21 @@ of the import statement, up to and including the final dot.
...
+
+The rule accepts the following parameters:
+
+- `rewrite.imports.groups`: defines several sets of regular expressions; each
+  set defines a single group, and the groups are output in the order they
+  are configured (imports not matching any of the regexes will form their own
+  group at the end). Regular expressions are applied to the entire parent domain
+  of the import statement, up to and including the final dot.
+- `rewrite.imports.contiguousGroups` (since v3.0.2):
+  - if `no` (default), grouping will happen on all imports within the same
```
Comment (tgodzik) [inline docs/configuration.md:2673]: I would honestly make it a default and not even add a setting for this. There is a high chance that `val` or something else between imports is there on purpose.

Comment (kitbellew) [reply docs/configuration.md:2673]: i can make it default but in our codebase, we prefer to take imports spread out around the file and put them in one place. users can define their vals and then import everything.

Comment (tgodzik) [reply docs/configuration.md:2673]: Ach, ok, it makes sense. But I think the default should be to not touch it, that's the safer side.

Comment (kitbellew) [reply docs/configuration.md]: changed the default

Comment (tgodzik) [reply docs/configuration.md]: Awesome thanks!

**Review by tgodzik — APPROVED** [review]

**Pull request merged**